### ISSUE TITLE
How can I change the initial prediction confidence threshold during training?

### ISSUE THREAD

**Issue by Potcasdace** [ultralytics/ultralytics#16662]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello, I wanted to find out the redundant boxes that network have predicted during training, which is to show the boxes with confidence score lower than 0.25, but I have met problem when changing the based values. I was wondering the first definition of  initial value of predict conf. When we modify or set the conf in default.yaml, which we reset the initial value of validation conf, which could filtered out boxes with conf lower than that threshold. I have make a little essays like using a defined value "conf_thres=0.0001" as a replacement of self.args.conf in ultalytics/models/yolo/detect/postprocess as shown in the figure. And conf=0.0002 was setted in the hyperparameters.
However, still got no boxes with conf lower than 0.25 in the plotted pictures.
Advanced great thanks for tour reply!!
<img width="571" alt="conf_thres" src="https://github.com/user-attachments/assets/5772b339-d0aa-4206-a4eb-fc60c18bc153">

<img width="671" alt="print" src="https://github.com/user-attachments/assets/19f8d3e3-73b9-4d43-a2ec-536c8aea4851">

[val_batch1_pred](https://github.com/user-attachments/assets/0185ddcd-a9fd-4105-9deb-e3300da3e708)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Potcasdace, thank you for your interest in Ultralytics 🚀! We recommend checking out our [Docs](https://docs.ultralytics.com) to explore various [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples that might provide insights on your query.

It seems like you're exploring how to adjust prediction confidence thresholds. To assist us in better understanding the situation, if this is a 🐛 Bug Report, we kindly ask you to provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/).

If this is a custom training ❓ Question, additional details such as dataset examples, training logs, and confirmation of aligning with our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) would be extremely helpful.

Connect with our vibrant community for quick support or detailed discussions. We're on [Discord](https://ultralytics.com/discord) 🎧, [Discourse](https://community.ultralytics.com), and [Subreddit](https://reddit.com/r/ultralytics).

## Upgrade

Ensure your environment is up-to-date with the latest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) using [**Python>=3.8**](https://www.python.org/) and [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO operates well in several environments with dependencies like [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn):

- **Notebooks** with GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud**, **Amazon** AMI and **Docker**. See related [Quickstart Guides](https://docs.ultralytics.com/yolov5/environments/)

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

Our [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) ensures functionality across various platforms.

This is an automated response, and an Ultralytics engineer will also assist you soon. Thank you for your patience and understanding! 😊

**Comment by Potcasdace** [comment]
Oh no, I mean i have changed the predict model, sorry for such a stupid question!

**Comment by glenn-jocher** [comment]
@Potcasdace no worries at all! If you have any more questions or need further assistance, feel free to ask.
